app/domain/authentication/authn_azure/validate_application_identity.rb

```diff
@@ -0,0 +1,102 @@
+require 'command_class'
+
+module Authentication
+  module AuthnAzure
+
+    Log = LogMessages::Authentication
+    Err = Errors::Authentication::AuthnAzure
+    # Possible Errors Raised: RoleNotFound, InvalidApplicationIdentity
+
+    ValidateApplicationIdentity = CommandClass.new(
+        dependencies: {
+            role_class:                 Role,
+            resource_class:             Resource,
+            application_identity_class: ApplicationIdentity,
+            logger:                     Rails.logger
+        },
+        inputs: %i(account service_id username xms_mirid oid)
+    ) do
+
+      def call
+        parse_xms_mirid
+        # compare xms_mirid with what is defined in annotations
+        validate_application_identity
+      end
+
+      private
+
+      # validate integrity of annotations
+      def application_identity
+        @application_identity ||= @application_identity_class.new(
+            role_annotations: role_annotations,
+            service_id: @service_id
+        )
+      end
+
+      def role_annotations
+        role.annotations
+      end
+
+      # convert host ID to full host ID, cucumber:host:somepolicy
+      # TODO For a Conjur user, an incorrect full id will be returned
+      def role_id
+        @role_id ||= @role_class.roleid_from_username(@account, @username)
+      end
+
+      def role
+        @role ||= @resource_class[role_id]
+        raise SecurityErr::RoleNotFound(role_id) if @username.nil?
+        return @role
+      end
+
+      # TODO add parsing format to check xms mirid is in proper format
+      # xms_mirid is a term in Azure to define a claim that describes the resource that holds the encoding of the instance's
+      # subscription, resource group, and provider identity needed for authorization. xms_mirid is one of the fields in
+      # the JWT token. This function will extract the relevant information from xms_mirid claim and populate a representative
+      # hash with the appropriate fields.
+      def parse_xms_mirid
+        # xms_mirid claim starts with an extra '/' so we will be ignoring it for parsing purposes
+        # ex: /subscription/<subscription_id> -> subscription/<subscription_id>
+        _, *field_split = @xms_mirid.split('/')
+        @xms_mirid_hash = Hash[field_split.each_slice(2).to_a]
```

Comment:
extract to method `xms_mirid_hash`?

Reply:
like so:
```
def xms_mirid_hash
   @xms_mirid_hash
end
```
?

Reply:
not at all. more like this:
```
# xms_mirid claim starts with an extra '/' so we will be ignoring it for parsing purposes
# ex: /subscription/<subscription_id> -> subscription/<subscription_id>
def xms_mirid_hash
        _, *field_split = @xms_mirid.split('/')
        Hash[field_split.each_slice(2).to_a]
end
```

Reply:
that is how it was before and it was mentioned that there is no value of this separated out without the function that holds this logic. Is this still the case?

Reply:
I don't recall what was here before. Let's stick with extracting this to a method.